instrumentation/servlet/servlet-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/v3_0/Servlet3OutputStreamWriteBytesAdvice.java

```diff
@@ -0,0 +1,26 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+package io.opentelemetry.javaagent.instrumentation.servlet.v3_0;
+
+import static io.opentelemetry.javaagent.instrumentation.servlet.v3_0.snippet.Injection.getInjectionObject;
+
+import io.opentelemetry.javaagent.instrumentation.servlet.v3_0.snippet.InjectionObject;
+import java.io.IOException;
+import javax.servlet.ServletOutputStream;
+import net.bytebuddy.asm.Advice;
+
+public class Servlet3OutputStreamWriteBytesAdvice {
+
+  @Advice.OnMethodEnter(skipOn = Advice.OnDefaultValue.class, suppress = Throwable.class)
+  public static boolean methodEnter(
+      @Advice.This ServletOutputStream servletOutputStream,
+      @Advice.Argument(value = 0, readOnly = false) byte[] write)
+      throws IOException {
+    InjectionObject obj = getInjectionObject(servletOutputStream);
+    if (obj.injected() || obj.characterEncoding == null) {
```

Comment:
use  obj.characterEncoding to make sure the servletOutputStream is the "wrapped" one, or the one calling by other library. (Alternative: add a new variable called 'wrapped' inside InjectionObject to make it clearer)

Reply:
could move this condition inside of `stringInjection`